### PR TITLE
Add request header idp

### PR DESCRIPTION
**Pull request by jcaianirh**
Add jira story: https://jira.coreos.com/browse/CONSOLE-1434

Screen Shot 2019-05-13 at 4 01 27 PM

cc: @bmignano

### CODE REVIEW

File: frontend/public/components/cluster-settings/request-header-idp-form.tsx [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+            aria-describedby="login-url-help" />
+          <div className="help-block" id="login-url-help">
+            The URL to redirect unauthenticated requests from OAuth clients which expect WWW-Authenticate challenges.
+          </div>
+        </div>
+        <div className="co-form-section__separator" />
+        <h3>More Options</h3>
+        <IDPCAFileInput value={caFileContent} onChange={this.caFileChanged} isRequired />
+        <ListInput label="Client Common Names" onChange={this.clientCommonNamesChanged} />
+        <ListInput label="Headers" onChange={this.headersChanged} />
```
Comment (spadgett) [inline frontend/public/components/cluster-settings/request-header-idp-form.tsx]: I believe this is required, @enj?

File: frontend/public/components/cluster-settings/request-header-idp-form.tsx [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+            onChange={this.loginURLChanged}
+            value={loginURL}
+            id="login-url"
+            aria-describedby="login-url-help" />
+          <div className="help-block" id="login-url-help">
+            The URL to redirect unauthenticated requests from OAuth clients which expect WWW-Authenticate challenges.
+          </div>
+        </div>
+        <div className="co-form-section__separator" />
+        <h3>More Options</h3>
```
Comment (spadgett) [inline frontend/public/components/cluster-settings/request-header-idp-form.tsx]: All of these options could use descriptions.

File: frontend/public/components/cluster-settings/request-header-idp-form.tsx [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+        <title>{title}</title>
+      </Helmet>
+      <form onSubmit={this.submit} name="form" className="co-m-pane__body-group co-m-pane__form">
+        <h1 className="co-m-pane__heading">{title}</h1>
+        <p className="co-m-pane__explanation">
+          Use request header to identify users from request header values. It is typically used in combination with an authenticating proxy, which sets the request header value.
+        </p>
+        <IDPNameInput value={name} onChange={this.nameChanged} />
+        <div className="co-form-section__separator" />
+        <h3>URLs</h3>
```
Comment (spadgett) [inline frontend/public/components/cluster-settings/request-header-idp-form.tsx]:
```jsx
        <h3>URLs</h3>
        <p className="co-m-pane__explanation">
          At least one URL must be provided.
        </p>
```

**Comment by jcaianirh** [comment]
@spadgett Added code to address review comments. Will add hep text to list input in a follow on pr as discussed.

File: frontend/public/components/cluster-settings/request-header-idp-form.tsx [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+          <div className="help-block" id="login-url-help">
+            The URL to redirect unauthenticated requests from OAuth clients which expect WWW-Authenticate challenges.
+          </div>
+        </div>
+        <div className="co-form-section__separator" />
+        <h3>More Options</h3>
+        <IDPCAFileInput value={caFileContent} onChange={this.caFileChanged} isRequired />
+        <ListInput label="Client Common Names" onChange={this.clientCommonNamesChanged} />
+        <div className="help-block">
+          An optional list of common names to require a match from.
```
Comment (spadgett) [inline frontend/public/components/cluster-settings/request-header-idp-form.tsx]: This should be an optional prop to the `ListInput` component. Then we can set the right `aria-describedby` headers for accessibility. Also we need to fix up the spacing. I'd suggest putting the description either immediately below the input or between the label and input.

Comment (jcaianirh) [reply frontend/public/components/cluster-settings/request-header-idp-form.tsx]: @spadgett yeah i thought you said to add this in the follow on pr?

Comment (spadgett) [reply frontend/public/components/cluster-settings/request-header-idp-form.tsx]: If we do it as a follow on, I'd take the descriptions out of this PR.

Comment (jcaianirh) [reply frontend/public/components/cluster-settings/request-header-idp-form.tsx]: ok

Comment (jcaianirh) [reply frontend/public/components/cluster-settings/request-header-idp-form.tsx]: @spadgett removed help until follow on

**Pull request merged**